07-regression1.Rmd

```diff
@@ -312,90 +327,99 @@ sacr_tunek_plot <- ggplot(sacr_results, aes(x = neighbors, y = mean)) +
   geom_line() +
   labs(x = "Neighbours", y = "RMSPE")
 
+kmin <- sacr_min %>% pull(neighbors)
+
 sacr_tunek_plot
 ```
 
-What is happening here? To visualize the effect of different settings of $K$ on the
-regression model, we will plot the predicted values for house price from
-our K-NN regression models for 6 different values for $K$.  For each model, we
-predict a price for every possible home size across the range of home sizes we
-observed in the data set (here 500 to 4250 square feet) and we plot the
-predicted prices as a blue line:
+Figure \@ref(fig:07-howK) visualizes the effect of different settings of $K$ on the
```

Comment:
I think we need to justify here why we are plotting the predictions on the training data. Students asked about why we were doing this when I taught the course last. We probably also need to do this in the previous chapter when we discuss under and overfitting for classification as well.

Reply:
changed the language to say that we are predicting across a range